contributingGuides/philosophies/ASYNC.md

```diff
@@ -0,0 +1,117 @@
+### Async vs. Sync Code
+
+Async code is everywhere in our app: API calls, storage access, background tasks, test scripts, GitHub Actions, and more. This document explains how and when to use sequential vs. parallel async flows, and why our rules exist.
+
+## Why this matters
+
+- Clarity: Async/await makes inherently sequential logic easier to read and review.
+- Performance: Parallelizing independent work avoids unnecessary delays.
+- Consistency: Shared rules make it easier for contributors inside and outside Expensify to write reliable code.
+
+## Rules
+
+### - Use async/await for sequential flows
+   When order matters, `async/await` expresses intent in a clear, linear style.
+   Example: Upload a file → Parse it → Save results.
+
+    ```ts
+    const uploaded = await uploadFile(file);
+    const parsed = await parseReceipt(uploaded.url);
+    await saveExpense(parsed);
+    ```
+
+### - Independent steps MUST be run in parallel
+   If two operations don’t depend on each other, start them together.
+   Example: Fetch user data and permissions concurrently with `Promise.all`.
+
+    ```ts
+    const [user, permissions] = await Promise.all([
+    getUser(),
+    getPermissions(),
+    ]);
+    ```
+
+### - UI SHOULD launch independent async calls in parallel
+   Components should not wait for one API call before starting another unless there is a dependency. Rendering must never be blocked by network requests.
+
+   Refer to [DATA-BINDING.md](./DATA-BINDING.md) for full details.
+
+### - Sequential logic SHOULD be encapsulated outside the UI
+   If a flow really must happen in order, write it in `src/libs/` or an action/helper. The UI should call that as a single logical operation.
+
+
+### - `async/await `SHOULD be preferred over `.then/.catch`
+   Use `async/await` unless you’re:
+
+   * Wrapping callback-based APIs.
+
+  ```ts
+  function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => {
+      setTimeout(resolve, ms);
+  });
+  }
+  ```
+
+   * Creating deferred promises (signals like “ready” or “loaded”).
+
+  ```ts
+  let resolveReady: () => void;
+
+  const isReady = new Promise<void>((resolve) => {
+    resolveReady = resolve;
+  });
+
+  // later in the code
+  resolveReady();
+  ```
+
+## More Examples
+
+### 1) Error handling
+
+Error handling style depends on context. If you’re handling a single async operation, `.catch()` is concise and effective.
+
+```ts
+// GOOD
+async function getData(url: string) {
+  const data = await fetch(url).catch(() => fetchFallback(url));
+  return process(data);
+}
```

Comment:
NAB: I would actually soften this example to preferred because some uses of try/catch for single promises are good enough.
```suggestion
// PREFERRED
async function getData(url: string) {
  const data = await fetch(url).catch(() => fetchFallback(url));
  return process(data);
}
```